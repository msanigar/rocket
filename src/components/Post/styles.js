import styled from 'styled-components';

export const Strong = styled.strong`
    color: red;
`;

export const Test = styled.h5`
    color: green;
    margin: 0.5rem;
`;