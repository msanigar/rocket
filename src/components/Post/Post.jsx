import React from 'react';
import { Strong } from './styles';

const decodeHTMLEntities = (text) => {
    const entities = [
        ['amp', '&'],
        ['apos', '\''],
        ['#x27', '\''],
        ['#x2F', '/'],
        ['#39', '\''],
        ['#47', '/'],
        ['lt', '<'],
        ['gt', '>'],
        ['nbsp', ' '],
        ['quot', '"']
    ];

    for (let i = 0, max = entities.length; i < max; ++i)
        text = text.replace(new RegExp('&'+entities[i][0]+';', 'g'), entities[i][1]);

    return text;
}

const Post = (props) => {
    const { data } = props.data;
    const r = '/r/';
    const reddit = 'http://reddit.com/';
    return(
        <div key={data.id} className="row post">
            <div className="column column-25">
                <a href={reddit + data.permalink}><img className="placeholder" src={data.preview ? decodeHTMLEntities(data.preview.images[0].source.url) : 'build/img/on_color_large.png'} /></a>
                <p className="center"><small>points: {data.score}</small></p>
            </div>
            <div className="column column-75">
                <p>
                <strong>{data.title}</strong>
                </p>
                <p><small>{r}{data.subreddit}</small> |
                <small>author: {data.author}</small> |
                <small><em><a href={reddit + data.permalink}>view on reddit.com</a></em></small></p>
            </div>
        </div>
    );
}

export default Post;
