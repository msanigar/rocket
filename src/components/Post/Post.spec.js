import React from 'React';
import Post from './Post';
import renderer from 'react-test-renderer';

it('Post should match snapshot', () => {
    const props = {
        data: {
            id: '1',
            permalink: '2',
            preview: {
                images: {
                    0: {
                        source: {
                            url: '3'
                        }
                    }
                }
            },
            score: '4',
            subreddit: '5',
            author: '6'
        }
    }
    const post = renderer
        .create(
            <Post data={props} />)
        .toJSON();
    expect(post).toMatchSnapshot();
});
