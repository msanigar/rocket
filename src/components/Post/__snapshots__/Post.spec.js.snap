// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post should match snapshot 1`] = `
<div
  className="row post"
>
  <div
    className="column column-25"
  >
    <a
      href="http://reddit.com/2"
    >
      <img
        className="placeholder"
        src="3"
      />
    </a>
    <p
      className="center"
    >
      <small>
        points: 
        4
      </small>
    </p>
  </div>
  <div
    className="column column-75"
  >
    <p>
      <strong />
    </p>
    <p>
      <small>
        /r/
        5
      </small>
       |
      <small>
        author: 
        6
      </small>
       |
      <small>
        <em>
          <a
            href="http://reddit.com/2"
          >
            view on reddit.com
          </a>
        </em>
      </small>
    </p>
  </div>
</div>
`;
